### PR TITLE
fix 56482 search should always work if instCode

### PR DESCRIPTION
**Pull request by OveBDev**
- fixed bug where projectCode became instCode for mc/commPack Search (named argument solves this, we should use this more)
- you can now do tag search on specified instCode, even if default instCode is not set (needed for echo3d)
- Clearer code: renamed to instCodeOrThrow

How to test: click changePlant with empty instCode, and try a tagSearch with instCode specified. It should do an online search now

### CODE REVIEW

File: src/workers/externalMcPacks.ts [hunk 1]
```diff
@@ -24,9 +25,10 @@ async function search(
     maxHits: number,
     tryToApplyFilter?: Filter<McPackDb>
 ): Promise<ResultArray<McPackDto>> {
+    const instCode = Settings.getInstCodeOrThrow(); //TODO - add instCode as argument
     return await _mcPacksSearchSystem.search(
         async () => inMemory.McPacks.search(searchText, maxHits, tryToApplyFilter),
-        () => mcPacksApi.search(searchText, maxHits, undefined, tryToApplyFilter?.projectName)
+        () => mcPacksApi.search({ searchText, maxHits, instCode, projectCode: tryToApplyFilter?.projectName })
```
Comment (herda1) [inline src/workers/externalMcPacks.ts:31]: The selected projectedName is set in EchoPedia, and can be set in the modules too. Is it needed to be here?

File: src/offlineSync/commPacksSyncer/commPacksApi.ts [hunk 2]
```diff
@@ -60,17 +60,16 @@ async function apiAllCommPackNos(instCode: string, abortSignal: AbortSignal): Pr
...
-    instCode = instCode ?? Settings.getInstCode();
-    let url = `${getApiBaseUrl()}/${instCode}/commPks`;
-    url += queryParameterOrEmpty('containsText', searchText, '?');
-    url += queryParameterOrEmpty('projectCodeContains', projectCode);
-    url += queryParameterOrEmpty('itemsPerPage', maxHits);
-    return commPacksApiFetcher.fetchAll(url, () => getMockedCommPacksString(100), abortSignal);
+async function apiSearchCommPacks(args: {
+    searchText: string;
+    maxHits: number;
+    instCode: string;
```
Comment (herda1) [inline src/offlineSync/commPacksSyncer/commPacksApi.ts:66]: does instCode need to be optional now? `instCode? `

File: src/workers/externalMcPacks.ts (hunk 1 above)
Comment (OveBDev) [reply src/workers/externalMcPacks.ts:31]: yes, we also want to filter on projectCode when we do online search

File: src/offlineSync/commPacksSyncer/commPacksApi.ts (hunk 2 above)
Comment (OveBDev) [reply src/offlineSync/commPacksSyncer/commPacksApi.ts:66]: instCode is required now, I think it's cleaner if the caller specifies the instCode

Comment (herda1) [reply src/offlineSync/commPacksSyncer/commPacksApi.ts:66]: Okay, then apiSearchMcPacks should behave the same way (below in comment).

File: src/offlineSync/mcPacksSyncer/mcPacksApi.ts [hunk 3]
```diff
@@ -50,17 +50,16 @@ async function apiUpdatedMcPacks(instCode: string, fromDate: Date, abortSignal:
...
-    instCode = instCode ?? Settings.getInstCode();
-    let url = `${getApiBaseUrl()}/${instCode}/mcPks`;
-    url += queryParameterOrEmpty('containsText', searchText, '?');
-    url += queryParameterOrEmpty('projectCodeContains', projectCode);
-    url += queryParameterOrEmpty('itemsPerPage', maxHits);
-    return mcPacksApiFetcher.fetchAll(url, () => getMockedMcPacksString(100), abortSignal);
+async function apiSearchMcPacks(args: {
+    searchText: string;
+    maxHits: number;
+    instCode?: string;
```
Comment (herda1) [inline src/offlineSync/mcPacksSyncer/mcPacksApi.ts:56]: here

**Review by herda1 — APPROVED** [review]

File: src/offlineSync/mcPacksSyncer/mcPacksApi.ts (hunk 3 above)
Comment (OveBDev) [reply src/offlineSync/mcPacksSyncer/mcPacksApi.ts]: good catch :)

**Pull request merged**